src/Microsoft.Tye.Core/DockerfileGenerator.cs

```diff
@@ -112,7 +112,7 @@ public static void ApplyContainerDefaults(ApplicationBuilder application, Dotnet
             }
 
             container.BuildImageName ??= "mcr.microsoft.com/dotnet/core/sdk";
-            container.BuildImageTag ??= "3.1";
+            container.BuildImageTag ??= project.TargetFrameworkName == "net" ? "5.0" : "3.1";
```

Comment:
What happens when net6.0 is out?

Reply:
Going to play around with setting this to the TargetFrameworkVersion. I think the idea before was that if someone has a 2.1 app, the 3.1 sdk would still be able to build it. However, now that we are dealing with 5.0, I'd rather not for all builds to happen with a preview SDK, hence the condition.